tier3/Dockerfile

```diff
@@ -30,6 +30,13 @@ RUN (cd /opt && \
     if [ "$(arch)" = x86_64 ]; then \
         mkdir /opt/lean && \
         curl --compressed -L "https://github.com/leanprover/lean4/releases/download/v4.0.0-m2/lean-4.0.0-m2-linux.tar.gz" | \
-        tar xz -C /opt/lean --strip-components=1; fi
+        tar xz -C /opt/lean --strip-components=1; fi && \
+    curl "https://packages.microsoft.com/config/debian/11/packages-microsoft-prod.deb" --output packages-microsoft-prod.deb && \
```

Comment:
If you move this up to the very start, we can batch the `apt-get update` and `install`.

Reply:
I've tried that but it seems the build process fails for some reason I don't fully understand. Leaving it the way it currently is seems to work though.